cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -458,7 +458,7 @@ public KafkaAssemblyOperator(Vertx vertx, boolean isOpenShift,
                 }
             }
             return withVoid(zkSetOperations.maybeRollingUpdate(zkDiffs.resource(),
-                    zkAncillaryCmChange || this.clusterCa.certRenewed() || this.clusterCa.certsRemoved()));
+                p -> zkAncillaryCmChange || this.clusterCa.certRenewed() || this.clusterCa.certsRemoved()));
```

Comment:
Especially in cases like this, where there's multiple possible causes for rolling the pod, I think we should log which of the reasons is causing the pod to be rolled. TBH this would be helpful even in single cause cases.

Reply:
@tombentley logging the reason is just above this code. Do you mean moving that part inside the predicate?

Reply:
Yes. I think it makes the log more readable to record the reason why we're starting a pod at the point that we're actually about to do it.